File: Java/civitas/src/civitas/SorpresaPorJugador.java
package civitas;

import java.util.ArrayList;


public class SorpresaPorJugador extends Sorpresa{
    private int valor;
    
    SorpresaPorJugador(int _valor, String _texto){
        super(_texto);
        valor = _valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            super.informe(actual, todos);
            SorpresaPagarCobrar sorpresilla = new SorpresaPagarCobrar(-valor*(todos.size()-1), "Pierdes " + valor + " euros para pagar a un jugador");
            SorpresaPagarCobrar recibeIngreso = new SorpresaPagarCobrar(valor, "Recibes dinero. Que suerte!");
            for(int i=0; i<todos.size(); i++){
                if(i==actual){
                    sorpresilla.aplicarAJugador(i, todos);
                }
                else{
                    recibeIngreso.aplicarAJugador(i, todos);
                }
            }
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa{ " + super.getTexto() + ", valor " + Integer.toString(valor) + " }";
                 
    }
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/sorpresa_por_jugador.rb
#encoding: utf­8
require_relative "sorpresa.rb"
module Civitas
class SorpresaPorJugador < Sorpresa
  attr_accessor :valor
  def initialize(_valor, _texto)
    super(_texto)
    @valor = _valor
  end
  
  def aplicar_a_jugador(_actual, _todos)
    if(jugador_correcto(_actual, _todos))
      informe(_actual, _todos)
      sorpresilla = SorpresaPagarCobrar.new(-@valor*(_todos.size()-1), "Pierdes dinero")
      recibe_ingreso = SorpresaPagarCobrar.new(@valor, "Ganas dinero")
      for i in 0..(_todos.size()-1)
        if(i == _actual)
          puts "PagarCobrar perder dinero"
          sorpresilla.aplicar_a_jugador(_actual, _todos)
        else
          puts "PagarCobrar ganar dinero"
          recibe_ingreso.aplicar_a_jugador(_actual, _todos)
        end
      end
    end
  end
  
  def to_ss()
    return @texto
  end
end
end
